lib/rucio/core/account.py

```diff
@@ -115,15 +117,14 @@ def del_account(account, *, session: "Session"):
         models.Account.status == AccountStatus.ACTIVE
     )
     try:
-        account = session.execute(query).scalar_one()
+        account_result = session.execute(query).scalar_one()
+        account_result.update({'status': AccountStatus.DELETED, 'deleted_at': datetime.utcnow()})
     except exc.NoResultFound:
         raise exception.AccountNotFound('Account with ID \'%s\' cannot be found' % account)
 
-    account.update({'status': AccountStatus.DELETED, 'deleted_at': datetime.utcnow()})
-
```

Comment:
The original statement `account = session.execute(query).scalar_one()` would lead to a type error, because the type of `account` (`InternalAccount`) was not compatible with the type returned by `session.execute(query).scalar_one()` (I don't remember what this type is at the moment). 

To avoid this, I changed the statemente so that the output of `session.execute(query).scalar_one()` would be stored in a new variable `account_result`. 

As this variable is defined within the scope of the `try` block, I had to move the `update` statement under the `try` block as well.

Reply:
I’m not sure I follow. The `try` block does not create a new scope.

```python
>>> print(foo)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'foo' is not defined
>>> try:
...     foo = 'foo'
... except Exception:
...     pass
... 
>>> print(foo)
foo
>>> 
```

Reply:
You're right - fixed